editor/plugins/org.fusesource.ide.wsdl2rest/Stash/SpringCamelContextFactory.java

```diff
@@ -0,0 +1,123 @@
+     * @throws IllegalStateException if the given URL does not contain a single context definition
+     */
+    public static SpringCamelContext createSingleCamelContext(URL contextUrl, ClassLoader classsLoader) throws Exception {
+        List<SpringCamelContext> list = createCamelContextList(new UrlResource(contextUrl), classsLoader);
+        IllegalStateAssertion.assertEquals(1, list.size(), "Single context expected in: " + contextUrl);
+        return list.get(0);
+    }
+
+    /**
+     * Create a {@link SpringCamelContext} list from the given URL
+     */
+    public static List<SpringCamelContext> createCamelContextList(URL contextUrl, ClassLoader classsLoader) throws Exception {
+        return createCamelContextList(new UrlResource(contextUrl), classsLoader);
+    }
+
+    /**
+     * Create a {@link SpringCamelContext} list from the given bytes
+     */
+    public static List<SpringCamelContext> createCamelContextList(byte[] bytes, ClassLoader classsLoader) throws Exception {
+        return createCamelContextList(new ByteArrayResource(bytes), classsLoader);
+    }
+
+    private static List<SpringCamelContext> createCamelContextList(Resource resource, ClassLoader classLoader) throws Exception {
+        
+        if (classLoader == null) 
+            classLoader = SpringCamelContextFactory.class.getClassLoader();
+        
+        GenericApplicationContext appContext = new GenericApplicationContext();
+        appContext.setClassLoader(classLoader);
+        
+        XmlBeanDefinitionReader xmlReader = new XmlBeanDefinitionReader(appContext) {
+            @Override
+            protected NamespaceHandlerResolver createDefaultNamespaceHandlerResolver() {
+                NamespaceHandlerResolver defaultResolver = super.createDefaultNamespaceHandlerResolver();
+                return new CamelNamespaceHandlerResolver(defaultResolver);
+            }
+        };
+        xmlReader.loadBeanDefinitions(resource);
+
+        SpringCamelContext.setNoStart(true);
+        appContext.refresh();
+
+        List<SpringCamelContext> result = new ArrayList<>();
+        for (String name : appContext.getBeanNamesForType(SpringCamelContext.class)) {
+            result.add(appContext.getBean(name, SpringCamelContext.class));
+        }
+
+        return Collections.unmodifiableList(result);
+    }
+
+    private static class CamelNamespaceHandlerResolver implements NamespaceHandlerResolver {
+
+        private final NamespaceHandlerResolver delegate;
+        private final NamespaceHandler camelHandler;
+
+        CamelNamespaceHandlerResolver(NamespaceHandlerResolver delegate) {
+            this.delegate = delegate;
+            this.camelHandler = new CamelNamespaceHandler();
+            this.camelHandler.init();
+        }
+
+        @Override
+        public NamespaceHandler resolve(String namespaceUri) {
+            if ("http://camel.apache.org/schema/spring".equals(namespaceUri)) {
```

Comment:
the constant org.fusesource.ide.foundation.core.xml.namespace.SpringNamespaceHandler.NAMESPACEURI_CAMEL_SPRING can be used

Reply:
Unneeded file. Removed from project.